lib/internal/url.js

```diff
@@ -817,6 +817,9 @@ class URLSearchParams {
     if (arguments.length < 1) {
       throw new TypeError('The `callback` argument needs to be specified');
     }
+    if (typeof callback !== 'function') {
+      throw new TypeError('The `callback` argument must be a function');
```

Comment:
our standard way of writing this would be `TypeError('"callback" argument must be a function')`